src/Lykke.Service.PayInternal.AzureRepositories/MerchantGroup/MerchantGroupEntity.cs

```diff
@@ -29,27 +30,34 @@ public MerchantGroupEntity(string partitionKey, string rowKey)
             PartitionKey = partitionKey;
             RowKey = rowKey;
         }
+        public static string GenerateRowKey()
+        {
+            return Guid.NewGuid().ToString();
+        }
+        public static string GeneratePartitionKey(string ownerId)
+        {
+            return ownerId;
+        }
         public static class ByOwner
         {
-            public static string GeneratePartitionKey(string merchantId)
+            public static string GeneratePartitionKey(string groupId)
             {
-                return merchantId;
+                return groupId;
             }
 
-            public static string GenerateRowKey(string id = null)
+            public static string GenerateRowKey(string ownerId)
             {
-                return id ?? Guid.NewGuid().ToString();
+                return ownerId;
             }
-
-            public static MerchantGroupEntity Create(IMerchantGroup merchantGroup)
+            public static string GenerateRowKey()
             {
-                var entity = new MerchantGroupEntity
-                {
-                    PartitionKey = GeneratePartitionKey(merchantGroup.OwnerMerchantId),
-                    RowKey = GenerateRowKey()
-                };
-
-                return Mapper.Map(merchantGroup, entity);
+                return "GroupIdIndex";
+            }
+            public static AzureIndex Create(MerchantGroupEntity entity)
+            {
+                return AzureIndex.Create(
+                    GeneratePartitionKey(entity.OwnerId),
```

Comment:
If you use ownerId as partition key for index, how are you going to store indexes for multiple groups with same owner? This code will not work and you will get an exception.